backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal_sap/process/SAP_GLJournal_ReverseDocument.java

```diff
@@ -74,12 +72,15 @@ public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNul
 	@Override
 	protected String doIt()
 	{
-		final SAPGLJournal createdJournal = glJournalService.copy(SAPGLJournalCopyRequest.builder()
+		final SAPGLJournal createdJournal = glJournalService.reverse(SAPGLJournalCopyRequest.builder()
 																			  .sourceJournalId(SAPGLJournalId.ofRepoId(getRecord_ID()))
 																			  .dateDoc(dateDoc)
-																			  .negateAmounts(negateAmounts)
 																			  .build());
 
+		final I_SAP_GLJournal sourceRecord = InterfaceWrapperHelper.load(SAPGLJournalId.ofRepoId(getRecord_ID()), I_SAP_GLJournal.class);
```

Comment:
loading a record outside of repo is not OK.

IMHO setting the DocStatus shall be the job of glJournalService.reverse too

Reply:
correction done via glJournalRepository